### PR TITLE
Feature: Support unix domain socket directory configuration

### PR DESCRIPTION
**Pull request by dengqian**
### Issues associated with this PR
mosn's reconfig.sock path is fixed,  to support hot upgrade between two containers, reconfigure unix domain socket path needs to be configurable.

### Solutions
add "uds_dir" filed, which is used to store *.sock files

### CODE REVIEW

File: pkg/mosn/init.go [hunk 1]
```diff
@@ -99,6 +99,10 @@ func initializeMetrics(config v2.MetricsConfig) {
 	}
 }
 
+func InitUDSdir (c *v2.MOSNConfig) {
```
Comment (taoyuanyuan) [inline pkg/mosn/init.go:102]: go fmt

Comment (taoyuanyuan) [reply pkg/mosn/init.go:102]: InitUDSDir

File: pkg/config/v2/config.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ type MOSNConfig struct {
 	RawAdmin             *Admin               `json:"admin,omitempty"`             // admin
 	Debug                PProfConfig          `json:"pprof,omitempty"`
 	Pid                  string               `json:"pid,omitempty"`                 // pid file
+	UDSdir               string               `json:"uds_dir,omitempty"`             // unix domain socket directory
```
Comment (taoyuanyuan) [inline pkg/config/v2/config.go:40]: UDSDir

File: pkg/mosn/init.go [hunk 3]
```diff
@@ -99,6 +99,10 @@ func initializeMetrics(config v2.MetricsConfig) {
 	}
 }
 
+func InitUDSDir (c *v2.MOSNConfig) {
```
Comment (taoyuanyuan) [inline pkg/mosn/init.go:102]: delete the space between func and args.

File: pkg/types/config.go [hunk 4]
```diff
@@ -77,3 +77,23 @@ end:
 	os.MkdirAll(MosnLogBasePath, 0755)
 	os.MkdirAll(MosnConfigPath, 0755)
 }
+
+
+func InitUDSDir(dir string){
```
Comment (nejisama) [inline pkg/types/config.go:82]: 是不是应该把调用做到InitDefaultPath里，如果有配置执行这个，没有的话，就用InitPath里默认那个。

**Comment by nejisama** [comment]
需要考虑如果热升级时 路径配置一样的行为是什么

File: pkg/types/config.go [hunk 5]
```diff
@@ -64,16 +65,23 @@ func InitDefaultPath(path string) {
...
 	MosnPidDefaultFileName = MosnLogBasePath + string(os.PathSeparator) + "mosn.pid"
-
 	MosnConfigPath = config
 
-	ReconfigureDomainSocket = MosnConfigPath + string(os.PathSeparator) + "reconfig.sock"
-	TransferConnDomainSocket = MosnConfigPath + string(os.PathSeparator) + "conn.sock"
-	TransferStatsDomainSocket = MosnConfigPath + string(os.PathSeparator) + "stats.sock"
-	TransferListenDomainSocket = MosnConfigPath + string(os.PathSeparator) + "listen.sock"
-	TransferMosnconfigDomainSocket = MosnConfigPath + string(os.PathSeparator) + "mosnconfig.sock"
+	if UDSDir != "" {
```
Comment (taoyuanyuan) [inline pkg/types/config.go:70]: 当udsdir为空的时候，MosnUDSPath需要赋值

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by nejisama — APPROVED** [review]

**Pull request merged**